### ISSUE TITLE
NPE in HBaseClient.fetchEntity() when no results are available

### ISSUE THREAD

**Issue by monkegoist** [Impetus/kundera#726]
Null-check is missing in `fetchEntity()` method:

``` java
results = handler.readData(tableName, subEntityMetadata.getEntityClazz(), subEntityMetadata, 
    rowId, subEntityMetadata.getRelationNames(), filter, columns);
// Result will not be empty for match sub entity.
if (!results.isEmpty()) { <-- NPE here
    break;
}
```

This happens when query doesn't return any results (for example, I try to find user with non-existent name). See `HBaseDataHandler#onRead()`: `output` is null, `results` is empty.

Reproduced in 2.15.1, but as far as I can see, this is also actual for 2.16.

**Comment by pragalbhgarg** [comment]
@monkegoist 
The part of code which you mentioned is for the JPA Inheritence implementation which is yet under development for kundera-hbase-v2. Hence, you might be facing such issues. We are actively working on it and will fix all such issues as soon as possible. Thanks for letting us know about the bug.

**Comment by monkegoist** [comment]
@pragalbhgarg I see, thanks for your reply. Any insights on when you're planning to release fix version? So far, I just added null check and this seems to fix the issue.

**Comment by pragalbhgarg** [comment]
@monkegoist 
We are planning on a patch release by mid April. Also, we have fixed the NPE in the trunk and will fix other related issues soon. We will notify you as we do that.

**Comment by chhavigangwal** [comment]
Fixed with Kundera -2.17 release.
